docs/workflow-messaging.md

```diff
@@ -0,0 +1,77 @@
+
+We can have more than one WorkflowProcessor service picking up a job, but we don't want more than one environment synchronising with the same DLCS space.
+So DDS production and test each have their own queue(s) subscribed to both Goobi and born-digital topics; prod writes to space 5 and test writes to space 7 (this is NEW)
+DDS staging writes to DLCS space 6, as now.
+
+In normal operation the `Test` environment does not listen to queues, or it throws away messages, etc - we control this through a flag so we can turn test on went we want to observe it. Even when test is running, its workflowProcessor is not always acting on the messages it picks up; it needs a control switch (e.g., a sentinel S3 key) to act on the messages.
+
+Stage should listen to Goobi stage and Storage stage, and sync with DLCS, all the time it's running (assuming much less traffic comes through stage, and the traffic that does come through still needs to be looked at via IIIF/DLCS).
+
+Dev environments, whether local or ad-hoc deployed, can use a separate set of queues just for dev, opting in to act on the messages. Or they could use the Test queues.
+
+So:
+
+SNS Topics ()
+
+- born-digital-bag-notifications-prod*
+- digitised-bag-notifications-prod
+- born-digital-bag-notifications-staging*
+- digitised-bag-notifications-staging
+
+SQS Queues
+
+- born-digital-bag-notifications-prod*
+- digitised-bag-notifications-prod
+- born-digital-bag-notifications-staging*
+- digitised-bag-notifications-staging
+- born-digital-bag-notifications-test (subscribes to topic born-digital-bag-notifications-prod)
+- digitised-bag-notifications-test (subscribes to topic digitised-bag-notifications-prod)
+- born-digital-bag-notifications-dev
+- digitised-bag-notifications-dev
+
+*existing
+
+These last 2 subscribe to the staging storage topics by default but can be switched to subscribe to the prod storage topics if required. Or, if we don't mind a profusion of SQS, we could replace the last two by:
+
+- born-digital-bag-notifications-dev-prod-storage
+- digitised-bag-notifications-dev-prod-storage
+- born-digital-bag-notifications-dev-stage-storage
+- digitised-bag-notifications-dev-stage-storage
+
+... to give us independent queues that won't be getting picked up by any deployed WorkflowProcessor, for all combinations we'd want to debug / investigate.
+This requires 4 queues being kept up that are hardly ever used, though. Maybe **not** have these but just have the dev ones as in the previous list but have two extra _topics_:
+
+- born-digital-bag-notifications-dev
+- digitised-bag-notifications-dev
+
+Neither Goobi nor the storage service broadcast to these topics, we have to manually send to them, for whatever dev setup we are testing to pick up.
```

Comment:
I prefer this approach of having to manually publish to SNS topics for dev. SNS + SQS can all live in digrati account.